base/abstractarraymath.jl

```diff
@@ -330,89 +308,147 @@ julia> repeat([1 2; 3 4], inner=(2, 1), outer=(1, 3))
 ```
 """
 function repeat(A::AbstractArray; inner = nothing, outer = nothing)
-    return _repeat_inner_outer(A, inner, outer)
+    return _RepeatInnerOuter.repeat(A, inner=inner, outer=outer)
+end
+
+module _RepeatInnerOuter
+
+function repeat(arr; inner=nothing, outer=nothing)
+    check(arr, inner, outer)
+    arr, inner, outer = resolve(arr, inner, outer)
+    repeat_inner_outer(arr, inner, outer)
+end
+
+to_tuple(t::Tuple) = t
+to_tuple(x::Integer) = (x,)
+to_tuple(itr) = tuple(itr...)
+
+function pad(a, b)
+    N = max(length(a), length(b))
+    Base.fill_to_length(a, 1, Val(N)), Base.fill_to_length(b, 1, Val(N))
```

Comment:
No need for `Base.` qualifications inside of Base.

Reply:
This is using a somewhat clever technique of a `module _InternalImplementation` which I don't think we've done before, but it cleverly helps readability of the internals while preventing clashes with other parts of base and leaking private functions.  I like it.

Thus it does need the qualifications.

Reply:
Huh, okay.